Sources/Markdown/Interpretive Nodes/Aside.swift

```diff
@@ -170,51 +180,54 @@ public struct Aside {
             // Otherwise, default to a note aside.
             self.kind = .note
             self.content = Array(blockQuote.blockChildren)
             return
         }
-        self.kind = Kind(rawValue: String(initialText.string[..<firstColonIndex]))!
 
-        // Trim off the aside tag prefix.
-        let trimmedText = initialText.string[initialText.string.index(after: firstColonIndex)...].drop {
-            $0 == " " || $0 == "\t"
+        self.kind = Kind(rawValue: kindTag.tag)!
+        self.content = Array(kindTag.newBlockQuote.blockChildren)
+    }
+
+    /// Create an aside from a block quote if it contains a kind tag that matches the given requirement.
+    public init?(_ blockQuote: BlockQuote, tagRequirement: TagRequirement = .tagNotRequired) {
+        guard tagRequirement != .tagNotRequired else {
+            self = .init(blockQuote)
+            return
         }
-        initialText.string = String(trimmedText)
 
-        let newBlockQuote = initialText.parent!.parent! as! BlockQuote
-        self.content = Array(newBlockQuote.blockChildren)
+        guard let kindTag = blockQuote.parseAsideTag(tagRequirement: tagRequirement) else {
+            return nil
+        }
+
+        self.kind = Kind(rawValue: kindTag.tag)!
+        self.content = Array(kindTag.newBlockQuote.blockChildren)
     }
 }
 
 extension BlockQuote {
-    /// Conservatively checks the text of this block quote to see whether it can be parsed as an aside.
-    ///
-    /// Whereas ``Aside/init(_:)`` will use all the text before the first colon in the first line,
-    /// or else return an ``Aside`` with a ``Aside/Kind`` of ``Aside/Kind/note``,
-    /// this function will allow parsers to only parse an aside if there is a single-word aside
-    /// marker in the first line, and otherwise fall back to a plain ``BlockQuote``.
-    func isAside() -> Bool {
-        guard let initialText = self.child(through: [
+    func parseAsideTag(tagRequirement: Aside.TagRequirement) -> (tag: String, newBlockQuote: BlockQuote)? {
+        guard var initialText = self.child(through: [
             (0, Paragraph.self),
             (0, Text.self),
-        ]) as? Text else {
-            return false
+        ]) as? Text, initialText.string.contains(":") else {
+            return nil
         }
 
-        for character in initialText.string {
-            switch character {
-                case ":": return true  // encountered ":" before " "
-                case " ": return false // encountered " " before ":"
-                default:  continue
-            }
+        let firstColonIndex = initialText.string.firstIndex(of: ":")!
+        let kindTag = initialText.string[..<firstColonIndex]
+        let trimmedText = initialText.string[initialText.string.index(after: firstColonIndex)...].drop {
+            $0 == " " || $0 == "\t"
+        }
+        initialText.string = String(trimmedText)
+        let newBlockQuote = initialText.parent!.parent! as! BlockQuote
```

Comment:
just curious: when is `initialText.parent!.parent!` not the same as `self`?

Reply:
I feel like it should always be the same, but i copied the existing code over in case it was different. I'll translate it to use `self` instead and check that it still works.